elasticdl/pkg/ps/optimizer.go

```diff
@@ -86,6 +86,57 @@ func (opt *SGDOptimizer) InitOptimizer(pb *proto.Model) error {
 	return nil
 }
 
+// MomentumOptimizer struct
+type MomentumOptimizer struct {
+	BaseOptimizer
+	mu       float32
+	nesterov bool
+	v        *Model
+}
+
+// GetLR returns learning rate
+func (opt *MomentumOptimizer) GetLR() float32 {
+	return opt.lr
+}
+
+// NewMomentumOptimizer creates a Momentum optimizer instance
+func NewMomentumOptimizer(lr float32, mu float32, nesterov bool) *MomentumOptimizer {
+	var opt = MomentumOptimizer{
+		BaseOptimizer: BaseOptimizer{
+			lr: lr,
+		},
+		mu:       mu,
+		nesterov: nesterov,
+		v:        NewModel(),
+	}
+	opt.DenseKernel = func(grad *common.Tensor, param *common.Tensor, name string) error {
+		v := opt.v.GetDenseParameter(name)
+		return kernel.Momentum(grad, param, v, opt.mu, opt.nesterov, opt.GetLR())
+	}
+	opt.SparseKernel = func(grad *common.IndexedSlices, param *common.EmbeddingTable, name string) error {
+		v := opt.v.GetEmbeddingTable(name)
+		return kernel.SparseMomentum(grad, param, v, opt.mu, opt.nesterov, opt.GetLR())
+	}
+	opt.IndexedKernel = func(grad *common.IndexedSlices, param *common.Tensor, name string) error {
+		v := opt.v.GetDenseParameter(name)
+		return kernel.IndexedMomentum(grad, param, v, opt.mu, opt.nesterov, opt.GetLR())
+	}
+	return &opt
+}
+
+// InitOptimizer set v non-embedding of MomentumOptimizer
+func (opt *MomentumOptimizer) InitOptimizer(pb *proto.Model) error {
```

Comment:
So this never returns error?

Reply:
Yes, I will clean the unnecessary return value of `error`